gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -221,6 +221,16 @@ public String apply(KafkaTopic topic) {
             }
           }), state);
 
+      for (KafkaTopic topic : topics) {
+        if (topic.getTopicSpecificState().isPresent()) {
+          if (topicSpecificStateMap.containsKey(topic.getName())) {
+            topicSpecificStateMap.get(topic.getName()).addAll(topic.getTopicSpecificState().get());
```

Comment:
Use `addAllIfNotExist`? Because we don't want to override configurations specified by the user.

Reply:
Good point, changed it.